plugins/reach/lib/model/data/VendorLiveCaptionCatalogItem.php

```diff
@@ -12,19 +12,27 @@ public function applyDefaultValues()
 
 	public function getTaskJobData($object)
 	{
+		$latestEvent = null;
 		if($object instanceof LiveEntry && $object->hasCapability(LiveEntry::LIVE_SCHEDULE_CAPABILITY))
 		{
 			$events = ScheduleEventPeer::retrieveByTemplateEntryIdAndTypes($object->getId(), [ScheduleEventType::LIVE_STREAM]);
 
-			$data = new kScheduledVendorTaskData();
-			if(count($events))
-			{
-				$latestEvent = $events[count($events)-1];
-				$data->setEntryDuration($latestEvent->getDuration()*1000);
-				$data->setStartDate($latestEvent->getStartDate());
-				$data->setEndDate($latestEvent->getEndDate());
-				$data->setScheduledEventId($latestEvent->getId());
+			if (count($events)) {
+				$latestEvent = $events[count($events) - 1];
```

Comment:
I don't think this is good (know it was here before and still) - as you assume the last event is the one you want to add the data from but that may not be the case. I never review that original PR that insert it.
Anyway, if you move to use event that sent the LiveStreamScheduleEvent why not remove all this part of the object been the live entry?

Reply:
@david-winder-kaltura  if you re-use the same live event for additional events later on, why wouldn't you use the latest scheduled event?
I would be inclined to leave the code in case we would need to support live captioning on entry update events in the future, but will defer to @salma2222